qc_tests/streaks.py

```diff
@@ -93,102 +170,354 @@ def get_repeating_string_threshold(obs_var, config_dict, plots=False, diagnostic
+                    plots: bool = False, diagnostics: bool = False) -> None:
+    """
+    Flag years where more than expected fraction of data occurs in streaks,
+      but none/not many are long enough in themselves to trigger the repeating_value check
+      Themselves need to be a reasonable length (10) else this overflags
+
+    Use config file to read threshold values.  Then find streaks which exceed threshold.
+
+    :param MetVar obs_var: meteorological variable object
+    :param array times: array of times (usually in minutes)
+    :param str config_dict: configuration file to store critical values
+    :param MetVar wind_speed: need speeds to mask calm periods in wind_directions
+    :param bool plots: turn on plots
+    :param bool diagnostics: turn on diagnostic output
+    
+    """
+    years = np.array([t.year for t in times])
+    if plots:
+        # Needed for plotting only
+        masked_times = np.ma.array(times, mask=obs_var.data.mask)
+    
+    this_var = prepare_obs_var(obs_var, wind_speed=wind_speed)
+
+    # use mask rather than compress immediately, to ensure indexing works
+    flags = np.ma.array(["" for i in range(this_var.data.shape[0])],
+                        mask=this_var.data.mask)
+
+    # retrieve the threshold and store
+    try:
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Excess"]
+    except KeyError:
+        # no threshold set
+        get_excess_streak_threshold(this_var, years, config_dict,
+                                    plots=plots, diagnostics=diagnostics)
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Excess"]
+
 
-    # more than expected number of short strings during a given period, but each individually not enough to set of main test
-    # HadISD - proportion of each year identified by strings, if >5x median, then remove these
+    if threshold == utils.MDI:
+        # No threshold obtainable, no need to continue the test
+        return
 
+    # Spin through each year, calculate proportion, flag comprising streaks if too high
+    for year in np.unique(years):
+        # set up indices for year, and flags array for year
+        locs, = np.nonzero(years == year)
+        year_flags = flags[locs]
+        unmasked, = np.nonzero(year_flags.mask == False)
+
+        if len(this_var.data[locs].compressed()) < 2:
+            # Escape if insufficient data
+            continue
+
+        # Not looking for distribution of streak lengths,
+        #  but proportion of obs identified as in a streak in each calendar year.
+        (year_repeated_streak_lengths,
+        grouped_diffs,
+        streaks) = utils.prepare_data_repeating_streak(this_var.data[locs].compressed(),
+                                                        diff=0, plots=plots,
+                                                        diagnostics=diagnostics)
+
+        long_enough_streaks, = np.where(year_repeated_streak_lengths >= MIN_STREAK_LENGTH_FOR_EXCESS_FREQUENCY)
+
```

Comment:
Use of np.where which changes when numpy 2.0 used?

Reply:
Done